### ISSUE TITLE
FailedPreconditionError on Adam optmizer

### ISSUE THREAD

**Issue by jacobwang123** [tflearn/tflearn#734]
Hi, love this library, awesome work.

Here's a bug one I run this code:
`config = tf.ConfigProto(log_device_placement=True, allow_soft_placement = True)
sess = tf.Session(config = config)
with tf.device('/gpu:0'):
    net = tflearn.input_data([None, 3])
    net = tflearn.embedding(net, input_dim=1593494, output_dim=256)
    net = tflearn.lstm(net, 256, dropout=0.8)
    net = tflearn.fully_connected(net, 3817, activation='softmax')
    net = tflearn.regression(net, optimizer='adam', learning_rate=0.001,\
                             loss='categorical_crossentropy', batch_size=256)
    sess.run(tf.global_variables_initializer())

model = tflearn.DNN(net, tensorboard_verbose=0, session=sess)
model.fit(trainX, trainY, validation_set=(validX, validY), show_metric=True, n_epoch=3)`

Error info:
FailedPreconditionError (see above for traceback): Attempting to use uninitialized value Embedding/W/Adam_1
	 [[Node: Embedding/W/Adam_1/read = Identity[T=DT_FLOAT, _class=["loc:@Embedding/W"], _device="/job:localhost/replica:0/task:0/cpu:0"](Embedding/W/Adam_1)]]

If you could help me one this, that would be great. Thank you!

**Comment by aymericdamien** [comment]
Thanks for your support :)
Due to some limitations, the embedding layer variables is set to be on device "cpu:0", so you may not alocate it to "gpu:0".
If you only have 1 gpu, you do not need to use "gpu:0", TensorFlow will automatically assign the compatible ops on GPU device.

**Comment by jacobwang123** [comment]
Got it, thank you for that :)
